controllers/application_controller.go

```diff
@@ -83,7 +115,12 @@ func (r *ApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		if apierrors.IsNotFound(err) {
 			argocdApp = nil
 		}
-
+		if argocdApp != nil && argocdApp.GetDeletionTimestamp() != nil {
+			logger.Info("argocd application is deleting. requeue the request")
+			return ctrl.Result{
+				Requeue: true,
```

Comment:
This is polling, I think it's better to use RequeueAfter with some waiting duration.

Reply:
`RequeAfter`: the backoff is constant.
`Reque`: the backoff is exponential.

I think `Reque` is more appropriate.